### ISSUE TITLE
GLONASS L2 sample config does not start

### ISSUE THREAD

**Issue by gaudima** [gnss-sdr/gnss-sdr#239]
Hello!
I just tried to launch sample glonass l2 config from this repo and it does not start. When I change this config to work with glonass l1 everything works as expected.

This seems to be some kind of a bug in gnuradio block connection, I tried to debug it but I dont have very much experience with gnuradio.

Here is config for glonass l2:

```
; This is a GNSS-SDR configuration file
; The configuration API is described at https://gnss-sdr.org/docs/sp-blocks/

[GNSS-SDR]

;######### GLOBAL OPTIONS ##################
GNSS-SDR.internal_fs_sps=6625000

;######### SIGNAL_SOURCE CONFIG ############
SignalSource.implementation=File_Signal_Source
SignalSource.filename=NT1065_GLONASS_L2_20160831_fs6625e6_60e3_schar_1m.bin ; <- PUT YOUR FILE HERE
SignalSource.item_type=ibyte
SignalSource.sampling_frequency=6625000
SignalSource.samples=0
SignalSource.dump=false;
SignalSource.dump_filename=/archive/signal_glonass.bin

;######### SIGNAL_CONDITIONER CONFIG ############
SignalConditioner.implementation=Signal_Conditioner
DataTypeAdapter.implementation=Ibyte_To_Complex
InputFilter.implementation=Pass_Through
InputFilter.item_type=gr_complex
Resampler.implementation=Pass_Through
Resampler.item_type=gr_complex

;######### CHANNELS GLOBAL CONFIG ############
Channel.signal=2G
Channels.in_acquisition=1
Channels_2G.count=5

;######### ACQUISITION GLOBAL CONFIG ############
Acquisition_2G.implementation=GLONASS_L2_CA_PCPS_Acquisition
Acquisition_2G.item_type=gr_complex
Acquisition_2G.threshold=0.0
Acquisition_2G.pfa=0.0001
Acquisition_2G.doppler_max=10000
Acquisition_2G.doppler_step=250
Acquisition_2G.dump=false;
Acquisition_2G.dump_filename=/archive/glo_acquisition.dat
;Acquisition_2G.coherent_integration_time_ms=1
;Acquisition_2G.max_dwells = 5

;######### TRACKING GLOBAL CONFIG ############
Tracking_2G.implementation=GLONASS_L2_CA_DLL_PLL_Tracking
Tracking_2G.item_type=gr_complex
Tracking_2G.early_late_space_chips=0.5
Tracking_2G.pll_bw_hz=20.0;
Tracking_2G.dll_bw_hz=2.0;
Tracking_2G.dump=false;
Tracking_2G.dump_filename=/archive/glo_tracking_ch_

;######### TELEMETRY DECODER GPS CONFIG ############
TelemetryDecoder_2G.implementation=GLONASS_L2_CA_Telemetry_Decoder

;######### OBSERVABLES CONFIG ############
Observables.implementation=Hybrid_Observables
Observables.dump=false;
Observables.dump_filename=/archive/glo_observables.dat

;######### PVT CONFIG ############
PVT.implementation=RTKLIB_PVT
PVT.positioning_mode=Single
PVT.output_rate_ms=100
PVT.display_rate_ms=500
PVT.trop_model=Saastamoinen
PVT.flag_rtcm_server=false
PVT.flag_rtcm_tty_port=false
PVT.rtcm_dump_devname=/dev/pts/1
PVT.rtcm_tcp_port=2101
PVT.rtcm_MT1019_rate_ms=5000
PVT.rtcm_MT1045_rate_ms=5000
PVT.rtcm_MT1097_rate_ms=1000
PVT.rtcm_MT1077_rate_ms=1000
PVT.rinex_version=2
```
The output with this config is:

```
linux; GNU C++ version 7.3.0; Boost_106501; UHD_003.010.003.000-0-unknown

Initializing GNSS-SDR v0.0.10 ... Please wait.
Logging will be written at "/tmp"
Use gnss-sdr --log_dir=/path/to/log to change that.
Processing file NT1065_GLONASS_L2_20160831_fs6625e6_60e3_schar_1m.bin, which contains 838860800 [bytes]
GNSS signal recorded time to be processed: 63.3092 [s]
E0114 18:09:40.868455 10194 gnss_flowgraph.cc:350] port number 5 exceeds max of 0
E0114 18:09:40.868515 10194 control_thread.cc:236] Unable to connect flowgraph
Total GNSS-SDR run time: 0.000268278 [seconds]
GNSS-SDR program ended.
```

This is config for glonass l1 (I changed only block implementations and signal type):

```
; This is a GNSS-SDR configuration file
; The configuration API is described at https://gnss-sdr.org/docs/sp-blocks/

[GNSS-SDR]

;######### GLOBAL OPTIONS ##################
GNSS-SDR.internal_fs_sps=6625000

;######### SIGNAL_SOURCE CONFIG ############
SignalSource.implementation=File_Signal_Source
SignalSource.filename=NT1065_GLONASS_L1_20160831_fs6625e6_60e3_schar_1m.bin ; <- PUT YOUR FILE HERE
SignalSource.item_type=ibyte
SignalSource.sampling_frequency=6625000
SignalSource.samples=0
SignalSource.dump=false;
SignalSource.dump_filename=/archive/signal_glonass.bin

;######### SIGNAL_CONDITIONER CONFIG ############
SignalConditioner.implementation=Signal_Conditioner
DataTypeAdapter.implementation=Ibyte_To_Complex
InputFilter.implementation=Pass_Through
InputFilter.item_type=gr_complex
Resampler.implementation=Pass_Through
Resampler.item_type=gr_complex

;######### CHANNELS GLOBAL CONFIG ############
Channel.signal=1G
Channels.in_acquisition=1
Channels_1G.count=5

;######### ACQUISITION GLOBAL CONFIG ############
Acquisition_1G.implementation=GLONASS_L1_CA_PCPS_Acquisition
Acquisition_1G.item_type=gr_complex
Acquisition_1G.threshold=0.0
Acquisition_1G.pfa=0.0001
Acquisition_1G.doppler_max=10000
Acquisition_1G.doppler_step=250
Acquisition_1G.dump=false;
Acquisition_1G.dump_filename=/archive/glo_acquisition.dat
;Acquisition_1G.coherent_integration_time_ms=1
;Acquisition_1G.max_dwells = 5

;######### TRACKING GLOBAL CONFIG ############
Tracking_1G.implementation=GLONASS_L1_CA_DLL_PLL_Tracking
Tracking_1G.item_type=gr_complex
Tracking_1G.early_late_space_chips=0.5
Tracking_1G.pll_bw_hz=20.0;
Tracking_1G.dll_bw_hz=2.0;
Tracking_1G.dump=false;
Tracking_1G.dump_filename=/archive/glo_tracking_ch_

;######### TELEMETRY DECODER GPS CONFIG ############
TelemetryDecoder_1G.implementation=GLONASS_L1_CA_Telemetry_Decoder

;######### OBSERVABLES CONFIG ############
Observables.implementation=Hybrid_Observables
Observables.dump=false;
Observables.dump_filename=/archive/glo_observables.dat

;######### PVT CONFIG ############
PVT.implementation=RTKLIB_PVT
PVT.positioning_mode=Single
PVT.output_rate_ms=100
PVT.display_rate_ms=500
PVT.trop_model=Saastamoinen
PVT.flag_rtcm_server=false
PVT.flag_rtcm_tty_port=false
PVT.rtcm_dump_devname=/dev/pts/1
PVT.rtcm_tcp_port=2101
PVT.rtcm_MT1019_rate_ms=5000
PVT.rtcm_MT1045_rate_ms=5000
PVT.rtcm_MT1097_rate_ms=1000
PVT.rtcm_MT1077_rate_ms=1000
PVT.rinex_version=2
```
And output with this config is:

```
linux; GNU C++ version 7.3.0; Boost_106501; UHD_003.010.003.000-0-unknown

Initializing GNSS-SDR v0.0.10 ... Please wait.
Logging will be written at "/tmp"
Use gnss-sdr --log_dir=/path/to/log to change that.
Processing file NT1065_GLONASS_L1_20160831_fs6625e6_60e3_schar_1m.bin, which contains 838860800 [bytes]
GNSS signal recorded time to be processed: 63.3092 [s]
Current receiver time: 1 s
Tracking of GLONASS L1 C/A signal started on channel 2 for satellite Glonass PRN 11 (Block 0)
Tracking of GLONASS L1 C/A signal started on channel 1 for satellite Glonass PRN 18 (Block -3)
Tracking of GLONASS L1 C/A signal started on channel 4 for satellite Glonass PRN 21 (Block 4)
Tracking of GLONASS L1 C/A signal started on channel 3 for satellite Glonass PRN 02 (Block -4)
Loss of lock in channel 1!
Tracking of GLONASS L1 C/A signal started on channel 0 for satellite Glonass PRN 01 (Block 1)
Current receiver time: 2 s
Tracking of GLONASS L1 C/A signal started on channel 1 for satellite Glonass PRN 12 (Block -1)
Loss of lock in channel 3!
Tracking of GLONASS L1 C/A signal started on channel 3 for satellite Glonass PRN 02 (Block -4)
Current receiver time: 3 s
Current receiver time: 4 s
Loss of lock in channel 4!
Tracking of GLONASS L1 C/A signal started on channel 4 for satellite Glonass PRN 21 (Block 4)
Current receiver time: 5 s
Current receiver time: 6 s
Current receiver time: 7 s
Current receiver time: 8 s
Current receiver time: 9 s
Current receiver time: 10 s
Current receiver time: 11 s
Current receiver time: 12 s
Current receiver time: 13 s
Current receiver time: 14 s
Current receiver time: 15 s
Current receiver time: 16 s
Current receiver time: 17 s
Current receiver time: 18 s
Current receiver time: 19 s
Current receiver time: 20 s
Current receiver time: 21 s
Current receiver time: 22 s
Current receiver time: 23 s
New GLONASS L1 GNAV almanac received in channel 1 from satellite Glonass PRN 12 (Block -1)
New GLONASS L1 GNAV almanac received in channel 2 from satellite Glonass PRN New GLONASS L1 GNAV almanac received in channel 11 (Block 0)
4 from satellite Glonass PRN 21 (Block 4)
New GLONASS L1 GNAV almanac received in channel 3 from satellite Glonass PRN 02 (Block -4)
New GLONASS L1 GNAV almanac received in channel 0 from satellite Glonass PRN 01 (Block 1)
Current receiver time: 24 s
Current receiver time: 25 s
New GLONASS L1 GNAV almanac received in channel 4 from satellite Glonass PRN 21 (Block 4)
New GLONASS L1 GNAV almanac received in channel 1 from satellite Glonass PRN 12 (Block -1)
New GLONASS L1 GNAV almanac received in channel 2 from satellite Glonass PRN 11 (Block 0)
New GLONASS L1 GNAV almanac received in channel 3 from satellite Glonass PRN 02 (Block -4)
New GLONASS L1 GNAV almanac received in channel 0 from satellite Glonass PRN 01 (Block 1)
Current receiver time: 26 s
Current receiver time: 27 s
Current receiver time: 28 s
Current receiver time: 29 s
Current receiver time: 30 s
Current receiver time: 31 s
Current receiver time: 32 s
Current receiver time: 33 s
Current receiver time: 34 s
Current receiver time: 35 s
Current receiver time: 36 s
Current receiver time: 37 s
Current receiver time: 38 s
Current receiver time: 39 s
New GLONASS L1 GNAV message received in channel 4: ephemeris from satellite Glonass PRN 17 (Block 4)
New GLONASS L1 GNAV message received in channel 4: UTC model parameters from satellite Glonass PRN 17 (Block 4)
New GLONASS L1 GNAV message received in channel 1: ephemeris from satellite Glonass PRN 16 (Block -1)
New GLONASS L1 GNAV message received in channel 1: UTC model parameters from satellite Glonass PRN 16 (Block -1)
New GLONASS L1 GNAV message received in channel 2: ephemeris from satellite Glonass PRN 15 (Block 0)
New GLONASS L1 GNAV message received in channel 3: ephemeris from satellite Glonass PRN 02 (Block -4)
New GLONASS L1 GNAV message received in channel 2: UTC model parameters from satellite Glonass PRN 15 (Block 0)
New GLONASS L1 GNAV message received in channel 3: UTC model parameters from satellite Glonass PRN 02 (Block -4)
Current receiver time: 40 s
First position fix at 2019-Apr-09 18:40:09 UTC is Lat = 40.0081 [deg], Long = -105.263 [deg], Height= 1609.1 [m]
Current receiver time: 41 s
The RINEX Navigation file header has been updated with UTC info.
Position at 2019-Apr-09 18:40:09.000000 UTC using 4 observations is Lat = 40.008055449 [deg], Long = -105.262592936 [deg], Height = 1609.099 [m]
Position at 2019-Apr-09 18:40:09.500000 UTC using 4 observations is Lat = 40.008100367 [deg], Long = -105.262644548 [deg], Height = 1574.130 [m]
Position at 2019-Apr-09 18:40:10.000000 UTC using 4 observations is Lat = 40.007986793 [deg], Long = -105.262567994 [deg], Height = 1628.737 [m]
Position at 2019-Apr-09 18:40:10.500000 UTC using 4 observations is Lat = 40.007939509 [deg], Long = -105.262589522 [deg], Height = 1667.524 [m]
Position at 2019-Apr-09 18:40:11.000000 UTC using 4 observations is Lat = 40.007982338 [deg], Long = -105.262533449 [deg], Height = 1651.411 [m]
Current receiver time: 42 s
Position at 2019-Apr-09 18:40:11.500000 UTC using 4 observations is Lat = 40.007902813 [deg], Long = -105.262496142 [deg], Height = 1706.117 [m]
Position at 2019-Apr-09 18:40:12.000000 UTC using 4 observations is Lat = 40.007920749 [deg], Long = -105.262537722 [deg], Height = 1697.337 [m]
Current receiver time: 43 s
Position at 2019-Apr-09 18:40:12.500000 UTC using 4 observations is Lat = 40.008011484 [deg], Long = -105.262585912 [deg], Height = 1684.137 [m]
New GLONASS L1 GNAV almanac received in channel 4 from satellite Glonass PRN 17 (Block 4)
New GLONASS L1 GNAV almanac received in channel 1 from satellite Glonass PRN 16 (Block -1)
New GLONASS L1 GNAV almanac received in channel 2 from satellite Glonass PRN 15 (Block 0)
New GLONASS L1 GNAV almanac received in channel 3 from satellite Glonass PRN 02 (Block -4)
Position at 2019-Apr-09 18:40:13.000000 UTC using 4 observations is Lat = 40.008011296 [deg], Long = -105.262557514 [deg], Height = 1664.833 [m]
Current receiver time: 44 s
Position at 2019-Apr-09 18:40:13.500000 UTC using 4 observations is Lat = 40.007927077 [deg], Long = -105.262522392 [deg], Height = 1681.138 [m]
Position at 2019-Apr-09 18:40:14.000000 UTC using 4 observations is Lat = 40.008011762 [deg], Long = -105.262605069 [deg], Height = 1692.587 [m]
Current receiver time: 45 s
Position at 2019-Apr-09 18:40:14.500000 UTC using 4 observations is Lat = 40.007851032 [deg], Long = -105.262532701 [deg], Height = 1765.138 [m]
Position at 2019-Apr-09 18:40:15.000000 UTC using 4 observations is Lat = 40.007959775 [deg], Long = -105.262575448 [deg], Height = 1721.053 [m]
Current receiver time: 46 s
Position at 2019-Apr-09 18:40:15.500000 UTC using 4 observations is Lat = 40.007908081 [deg], Long = -105.262588814 [deg], Height = 1708.424 [m]
Position at 2019-Apr-09 18:40:16.000000 UTC using 4 observations is Lat = 40.008026580 [deg], Long = -105.262615186 [deg], Height = 1683.687 [m]
Current receiver time: 47 s
Position at 2019-Apr-09 18:40:16.500000 UTC using 4 observations is Lat = 40.007903034 [deg], Long = -105.262615900 [deg], Height = 1742.917 [m]
New GLONASS L1 GNAV almanac received in channel 4 from satellite Glonass PRN 17 (Block 4)
New GLONASS L1 GNAV almanac received in channel 1 from satellite Glonass PRN 16 (Block -1)
New GLONASS L1 GNAV almanac received in channel 3 from satellite Glonass PRN 02 (Block -4)
New GLONASS L1 GNAV almanac received in channel 2 from satellite Glonass PRN 15 (Block 0)
Position at 2019-Apr-09 18:40:17.000000 UTC using 4 observations is Lat = 40.007980919 [deg], Long = -105.262627042 [deg], Height = 1679.855 [m]
Current receiver time: 48 s
Position at 2019-Apr-09 18:40:17.500000 UTC using 4 observations is Lat = 40.007990845 [deg], Long = -105.262585072 [deg], Height = 1676.177 [m]
Position at 2019-Apr-09 18:40:18.000000 UTC using 4 observations is Lat = 40.007906491 [deg], Long = -105.262622875 [deg], Height = 1722.066 [m]
Current receiver time: 49 s
Position at 2019-Apr-09 18:40:18.500000 UTC using 4 observations is Lat = 40.008006785 [deg], Long = -105.262618628 [deg], Height = 1663.039 [m]
Position at 2019-Apr-09 18:40:19.000000 UTC using 4 observations is Lat = 40.008229924 [deg], Long = -105.262772664 [deg], Height = 1563.122 [m]
Current receiver time: 50 s
Position at 2019-Apr-09 18:40:19.500000 UTC using 4 observations is Lat = 40.008153723 [deg], Long = -105.262683202 [deg], Height = 1588.285 [m]
Position at 2019-Apr-09 18:40:20.000000 UTC using 4 observations is Lat = 40.008182971 [deg], Long = -105.262707702 [deg], Height = 1553.728 [m]
Current receiver time: 51 s
Position at 2019-Apr-09 18:40:20.500000 UTC using 4 observations is Lat = 40.008030932 [deg], Long = -105.262641084 [deg], Height = 1628.964 [m]
New GLONASS L1 GNAV almanac received in channel 4 from satellite Glonass PRN 17 (Block 4)
New GLONASS L1 GNAV almanac received in channel 1 from satellite Glonass PRN 16 (Block -1)
New GLONASS L1 GNAV almanac received in channel 2 from satellite Glonass PRN 15 (Block 0)
New GLONASS L1 GNAV almanac received in channel 3 from satellite Glonass PRN 02 (Block -4)
Position at 2019-Apr-09 18:40:21.000000 UTC using 4 observations is Lat = 40.007857408 [deg], Long = -105.262561995 [deg], Height = 1731.008 [m]
Current receiver time: 52 s
Position at 2019-Apr-09 18:40:21.500000 UTC using 4 observations is Lat = 40.008043695 [deg], Long = -105.262629519 [deg], Height = 1671.141 [m]
Position at 2019-Apr-09 18:40:22.000000 UTC using 4 observations is Lat = 40.007961188 [deg], Long = -105.262539749 [deg], Height = 1685.236 [m]
Current receiver time: 53 s
Position at 2019-Apr-09 18:40:22.500000 UTC using 4 observations is Lat = 40.007936072 [deg], Long = -105.262571171 [deg], Height = 1693.769 [m]
Position at 2019-Apr-09 18:40:23.000000 UTC using 4 observations is Lat = 40.007841932 [deg], Long = -105.262595531 [deg], Height = 1748.630 [m]
Current receiver time: 54 s
Position at 2019-Apr-09 18:40:23.500000 UTC using 4 observations is Lat = 40.007882529 [deg], Long = -105.262656014 [deg], Height = 1718.515 [m]
Position at 2019-Apr-09 18:40:24.000000 UTC using 4 observations is Lat = 40.007820064 [deg], Long = -105.262525311 [deg], Height = 1750.685 [m]
Current receiver time: 55 s
Position at 2019-Apr-09 18:40:24.500000 UTC using 4 observations is Lat = 40.008015622 [deg], Long = -105.262675815 [deg], Height = 1674.410 [m]
New GLONASS L1 GNAV almanac received in channel 4 from satellite Glonass PRN 17 (Block 4)
New GLONASS L1 GNAV almanac received in channel 1 from satellite Glonass PRN 16 (Block -1)
New GLONASS L1 GNAV almanac received in channel 2 from satellite Glonass PRN 15 (Block 0)
New GLONASS L1 GNAV almanac received in channel 3 from satellite Glonass PRN 02 (Block -4)
Position at 2019-Apr-09 18:40:25.000000 UTC using 4 observations is Lat = 40.007727241 [deg], Long = -105.262475784 [deg], Height = 1810.331 [m]
Current receiver time: 56 s
Position at 2019-Apr-09 18:40:25.500000 UTC using 4 observations is Lat = 40.007971135 [deg], Long = -105.262657407 [deg], Height = 1666.582 [m]
Position at 2019-Apr-09 18:40:26.000000 UTC using 4 observations is Lat = 40.007801371 [deg], Long = -105.262531530 [deg], Height = 1758.940 [m]
Current receiver time: 57 s
Position at 2019-Apr-09 18:40:26.500000 UTC using 4 observations is Lat = 40.007892180 [deg], Long = -105.262556828 [deg], Height = 1728.390 [m]
Position at 2019-Apr-09 18:40:27.000000 UTC using 4 observations is Lat = 40.007841164 [deg], Long = -105.262544280 [deg], Height = 1740.505 [m]
Current receiver time: 58 s
Position at 2019-Apr-09 18:40:27.500000 UTC using 4 observations is Lat = 40.007918265 [deg], Long = -105.262533034 [deg], Height = 1704.752 [m]
Position at 2019-Apr-09 18:40:28.000000 UTC using 4 observations is Lat = 40.007856882 [deg], Long = -105.262551843 [deg], Height = 1703.399 [m]
Current receiver time: 59 s
New GLONASS L1 GNAV almanac received in channel 4 from satellite Glonass PRN 17 (Block 4)
Position at 2019-Apr-09 18:40:28.500000 UTC using 4 observations is Lat = 40.007997585 [deg], Long = -105.262672807 [deg], Height = 1633.038 [m]
New GLONASS L1 GNAV almanac received in channel 1 from satellite Glonass PRN 16 (Block -1)
New GLONASS L1 GNAV almanac received in channel 2 from satellite Glonass PRN 15 (Block 0)
New GLONASS L1 GNAV almanac received in channel 3 from satellite Glonass PRN 02 (Block -4)
Position at 2019-Apr-09 18:40:29.000000 UTC using 4 observations is Lat = 40.007912609 [deg], Long = -105.262634828 [deg], Height = 1682.858 [m]
Current receiver time: 1 min 0 s
Position at 2019-Apr-09 18:40:29.500000 UTC using 4 observations is Lat = 40.007856490 [deg], Long = -105.262548554 [deg], Height = 1698.130 [m]
Position at 2019-Apr-09 18:40:30.000000 UTC using 4 observations is Lat = 40.007953850 [deg], Long = -105.262650501 [deg], Height = 1662.934 [m]
Current receiver time: 1 min 1 s
Position at 2019-Apr-09 18:40:30.500000 UTC using 4 observations is Lat = 40.007889537 [deg], Long = -105.262604262 [deg], Height = 1705.754 [m]
Position at 2019-Apr-09 18:40:31.000000 UTC using 4 observations is Lat = 40.007871577 [deg], Long = -105.262508008 [deg], Height = 1718.888 [m]
Current receiver time: 1 min 2 s
Position at 2019-Apr-09 18:40:31.500000 UTC using 4 observations is Lat = 40.008015871 [deg], Long = -105.262625386 [deg], Height = 1665.707 [m]
Position at 2019-Apr-09 18:40:32.000000 UTC using 4 observations is Lat = 40.007862867 [deg], Long = -105.262580203 [deg], Height = 1726.337 [m]
Current receiver time: 1 min 3 s
Position at 2019-Apr-09 18:40:32.500000 UTC using 4 observations is Lat = 40.007888298 [deg], Long = -105.262627770 [deg], Height = 1723.143 [m]
Position at 2019-Apr-09 18:40:33.000000 UTC using 4 observations is Lat = 40.007979438 [deg], Long = -105.262640538 [deg], Height = 1669.685 [m]
Current receiver time: 1 min 4 s
Position at 2019-Apr-09 18:40:33.500000 UTC using 4 observations is Lat = 40.007985968 [deg], Long = -105.262663266 [deg], Height = 1669.484 [m]
Stopping GNSS-SDR, please wait!
Total GNSS-SDR run time: 8.683296 [seconds]
GNSS-SDR program ended.
```

**Comment by gaudima** [comment]
This should now be fixed after #240 merge.

**Comment by carlesfernandez** [comment]
Thanks for your contribution! PR https://github.com/gnss-sdr/gnss-sdr/pull/240 has been merged into `next`.

**Comment by carlesfernandez** [comment]
Please feel free to open a new issue if the problem persists.
